### PR TITLE
65313_user_suggest_to_admin

### PR DESCRIPTION
**Pull request by asifbuetcse**

### CODE REVIEW

File: app/src/Controller/SuggestionsController.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace App\Controller;
+
+use App\Controller\AppController;
+
+class SuggestionsController extends AppController
+{
+    public function isAuthorized($user)
+    {
+        if (isset($user['is_admin']) && $user['is_admin'] === true) {
+            if (in_array($this->request->action, ['index', 'view', 'delete'])) {
```
Comment (tuandv2611) [inline app/src/Controller/SuggestionsController.php]: Can you refactor this code for nice?

File: app/src/Controller/SuggestionsController.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function index()
+    {
+        $this->paginate = [
```
Comment (tuandv2611) [inline app/src/Controller/SuggestionsController.php]: with simple array, write them in a line
`$this->paginate = [ 'contain' => ['Users']];`

File: app/src/Controller/SuggestionsController.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        ];
+        $suggestions = $this->paginate($this->Suggestions);
+
+        $this->set(compact('suggestions'));
+        $this->set('_serialize', ['suggestions']);
+    }
+
+    public function view($id = null)
+    {
+        $suggestion = $this->Suggestions->get($id, [
```
Comment (tuandv2611) [inline app/src/Controller/SuggestionsController.php]: write it in a line

File: app/src/Template/Layout/default.ctp [hunk 4]
```diff
@@ -65,9 +65,14 @@ $cakeDescription = 'CakePHP: the rapid development php framework';
                             <li><a href="#">View All</a></li>
                         </ul>
                     </li>
+                    <li class="dropdown"><a class="dropdown-toggle" data-toggle="dropdown" href="#">Suggestions <span class="caret"></span></a>
```
Comment (tuandv2611) [inline app/src/Template/Layout/default.ctp]: break line
```
<li class="dropdown">
    <a
```

File: app/src/Template/Suggestions/view.ctp [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<div class="suggestions view large-9 medium-8 columns content">
+    <h3><?= h($suggestion->id) ?></h3>
+    <table class="table table-bordered">
+        <tr>
+            <th><?= __('User') ?></th>
+            <td><?= $suggestion->has('user') ? $this->Html->link($suggestion->user->id, [
```
Comment (tuandv2611) [inline app/src/Template/Suggestions/view.ctp]: need format

File: app/src/Template/Suggestions/index.ctp [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                <th><?= $this->Paginator->sort('user_id') ?></th>
+                <th><?= $this->Paginator->sort('product_name') ?></th>
+                <th class="actions"><?= __('Actions') ?></th>
+            </tr>
+        </thead>
+        <tbody>
+            <?php foreach ($suggestions as $suggestion): ?>
+            <tr>
+                <td><?= $this->Number->format($suggestion->id) ?></td>
+                <td><?= $suggestion->has('user') ? $this->Html->link($suggestion->user->id, [
```
Comment (tuandv2611) [inline app/src/Template/Suggestions/index.ctp]: need format again?

File: app/src/Controller/SuggestionsController.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        return parent::isAuthorized($user);
+    }
+
+    public function index()
+    {
+        $this->paginate = ['contain' => ['Users']];
+        $suggestions = $this->paginate($this->Suggestions);
+
+        $this->set(compact('suggestions'));
+        $this->set('_serialize', ['suggestions']);
```
Comment (tuandv2611) [inline app/src/Controller/SuggestionsController.php]: What do you use _serialize for?

File: app/src/Template/Layout/default.ctp [hunk 8]
```diff
@@ -65,9 +65,15 @@ $cakeDescription = 'CakePHP: the rapid development php framework';
                             <li><a href="#">View All</a></li>
                         </ul>
                     </li>
+                    <li class="dropdown">
+                        <a class="dropdown-toggle" data-toggle="dropdown" href="#">Suggestions <span class="caret"></span></a>
```
Comment (tuandv2611) [inline app/src/Template/Layout/default.ctp]: format again

File: app/src/Template/Suggestions/index.ctp [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                <th><?= $this->Paginator->sort('product_name') ?></th>
+                <th class="actions"><?= __('Actions') ?></th>
+            </tr>
+        </thead>
+        <tbody>
+            <?php foreach ($suggestions as $suggestion): ?>
+            <tr>
+                <td><?= $this->Number->format($suggestion->id) ?></td>
+                <td>
+                    <?= $suggestion->has('user') ? $this->Html->link($suggestion->user->id, [
```
Comment (tuandv2611) [inline app/src/Template/Suggestions/index.ctp]: if !$suggestion->has('user')? what's happen?

File: app/src/Template/Suggestions/view.ctp [hunk 10]
```diff
@@ -0,0 +1,23 @@
+<div class="suggestions view large-9 medium-8 columns content">
+    <h3><?= h($suggestion->id) ?></h3>
+    <table class="table table-bordered">
+        <tr>
+            <th><?= __('User') ?></th>
+            <td>
+                <?= $suggestion->has('user') ? $this->Html->link($suggestion->user->id, [
```
Comment (tuandv2611) [inline app/src/Template/Suggestions/view.ctp]: how do you check it?
I think that user always exist.

**Comment by tuandv2611** [comment]
OK

**Comment by TungPro** [comment]
OK

**Pull request merged**